<launch>
    <node name="rt_audio" pkg="rt_audio_ros" type="rt_audio">
        <rosparam>
            nb_microphones: 4
            card_name:      Kinect
            sample_rate:    16000
        </rosparam>
        <remap from="audio_stream" to="audio_stream_16k"/>
    </node>
    <node name="audio_upscaler" pkg="audio_tools" type="upscaler">
        <param name="rate" value="3"/>
        <remap from="audio_stream"    to="audio_stream_16k"/>
        <remap from="audio_stream_up" to="audio_stream"/>
    </node>

    <node name="manyears_node"
          pkg="manyears_ros"
          type="manyears_node_standalone"
          output="screen">
        <param name="config_file" 
               value="$(find manyears_ros)/data/kinect.mes"/>
        <rosparam>
            frame_id:    micro_center_link
            planar_mode: true
            pre_gain:    1.0
            # Geometric configuration for a Microsoft Kinect:
            microphones:
             - {pos: {x: 0.00, y: -0.113, z: 0.00}, gain: 1.0}
             - {pos: {x: 0.00, y:  0.036, z: 0.00}, gain: 1.0}
             - {pos: {x: 0.00, y:  0.076, z: 0.00}, gain: 1.0}
             - {pos: {x: 0.00, y:  0.113, z: 0.00}, gain: 1.0}
        </rosparam>
    </node>

    <node name="sound_pose" pkg="manyears_ros" type="position_exploitation">
        <remap from="/sources_info" to="/tracked_sources"/>
    </node>

    <!-- TEMP -->
    <node name="micro_tf" 
          pkg="tf" 
          type="static_transform_publisher"
          args="0 0 0 0 0 0 1 base_link micro_center_link 10"
    />

</launch>
